src/components/EditorWidgets/Date/DateControl.js

```diff
@@ -41,22 +42,33 @@ export default class DateControl extends React.Component {
   handleChange = datetime => {
     const { onChange } = this.props;
 
-    // Set the date only if the format is valid
-    if (this.isValidDate(datetime)) {
-      const formattedValue = moment(datetime).format(this.format);
+    /**
+     * Set the date only if it is valid.
+     */
+    if (!this.isValidDate(datetime)) {
+      return;
+    }
+
+    /**
+     * Produce a formatted string only if a format is set in the config.
+     * Otherwise produce a date object.
+     */
+    if (this.formatOutput) {
+      const formattedValue = moment(datetime).format(this.formatOutput);
       onChange(formattedValue);
+    } else {
+      onChange(datetime);
```

Comment:
Is it better to call `datetime.toDate()` here so that the formatters have a JS date object to work with instead of a Moment object?

Reply:
Nevermind, this may not work well for JSON: https://momentjs.com/docs/#/displaying/as-json/

Reply:
Yeah it matches previous behavior.